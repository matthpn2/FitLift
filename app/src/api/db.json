{
  "users": {
      "Swagmo996": {
          "name": "Teemo",
          "password": "teemoisgod69"
      },
      "kevanhn": {
          "name": "Kevan",
          "password": "asdfasdf1"
      }
  },
  "new_exercises" : {
    "SAMPLE_USER" : {
      "-L5GDNmo8uZxet9S7mE2" : {
        "reps" : 12,
        "timeStamp" : 1518407604,
        "type" : "Bicep Curls"
      },
      "-L5GDeJnjbOj1SkQRJK8" : {
        "reps" : 5,
        "timeStamp" : 1518407604,
        "type" : "Shoulder Press"
      },
      "-L5GDeeBKCPlTnm-yCiX" : {
        "reps" : 6,
        "timeStamp" : 1518407604,
        "type" : "Lateral Raises"
      }
    }
  },
  "exercises": {
      "Swagmo996": {
          "bicep_curls": {
              "last_set": "set2",
              "sets": {
                  "set1": {
                      "weight": 45,
                      "unit": "lbs",
                      "repetitions": 5,
                      "timestamp": 1516857363
                  },
                  "set2": {
                      "weight": 45,
                      "unit": "lbs",
                      "repetitions": 7,
                      "timestamp": 1516857963
                  }
              }
          },
          "overhead_press": {
              "last_set": "set4",
              "sets": {
                  "set1": {
                      "weight": 70,
                      "unit": "lbs",
                      "repetitions": 8,
                      "timestamp": 1516857363
                  },
                  "set2": {
                      "weight": 70,
                      "unit": "lbs",
                      "repetitions": 4,
                      "timestamp": 1516857963
                  },
                  "set3": {
                      "weight": 70,
                      "unit": "lbs",
                      "repetitions": 7,
                      "timestamp": 1516858963
                  },
                  "set4": {
                      "weight": 70,
                      "unit": "lbs",
                      "repetitions": 3,
                      "timestamp": 1516859963
                  }
              }
          },
          "chest_press": {
              "last_set": "set3",
              "sets": {
                  "set1": {
                      "weight": 55,
                      "unit": "lbs",
                      "repetitions": 12,
                      "timestamp": 1516857363
                  },
                  "set2": {
                      "weight": 55,
                      "unit": "lbs",
                      "repetitions": 11,
                      "timestamp": 1516857963
                  },
                  "set3": {
                      "weight": 55,
                      "unit": "lbs",
                      "repetitions": 9,
                      "timestamp": 1516858963
                  }
              }
          }
      },
      "kevanhn": {
          "bicep_curls": {
              "last_set": "set3",
              "sets": {
                  "set1": {
                      "weight": 40,
                      "unit": "lbs",
                      "repetitions": 5,
                      "timestamp": 1516857363
                  },
                  "set2": {
                      "weight": 40,
                      "unit": "lbs",
                      "repetitions": 7,
                      "timestamp": 1516857963
                  },
                  "set3": {
                      "weight": 40,
                      "unit": "lbs",
                      "repetitions": 7,
                      "timestamp": 1516857963
                  }
              }
          }
      }
  },
  "friends": {
      "Swagmo996": {
          "kevanhn": true
      },
      "kevanhn": {
          "Swagmo996": true
      }
  }
}