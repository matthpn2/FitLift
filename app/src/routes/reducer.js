import { NavigationActions } from 'react-navigation';
import { Navigation } from './index';

const initialState = Navigation.router.getStateForAction(
  Navigation.router.getActionForPathAndParams('login')
);

export default (state = initialState, action) => {
  switch (action.type) {
    case 'LOGIN_SUCCESS':
    case 'CREATE_USER_SUCCESS':
      return Navigation.router.getStateForAction(
        NavigationActions.navigate({ routeName: 'loggedIn' })
      );
    case 'logout':
      return Navigation.router.getStateForAction(
        NavigationActions.navigate({ routeName: 'login' })
      );
    case 'DISPLAY_EXERCISE_DAY':
      return Navigation.router.getStateForAction(
        NavigationActions.navigate({
          params: {
            day: action.day
          },
          routeName: 'exerciseDay'
        }),
        Navigation.router.getStateForAction(action, state)
      );
    default:
      return Navigation.router.getStateForAction(action, state);
  }
};
