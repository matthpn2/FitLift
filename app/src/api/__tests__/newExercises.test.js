import nock from 'nock';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { new_exercises as newExercises } from '../db.json';
import reducer, {
  TypeKeys,
  fetchNewExercises,
} from '../newExercises';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('async action creator tests', () => {
  it.skip('should fetch current exercises', () => {
    nock()
      .get('/new_exercises/SAMPLE_USER.json')
      .reply(200, newExercises);

    const store = mockStore();
    return store.dispatch(fetchNewExercises('SAMPLE_USER'))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});


describe('reducer tests', () => {
  it('TypeKeys.RECEIVE_CURRENT_EXERCISES should work', () => {
    expect(reducer(
      {},
      {
        data: newExercises,
        type: TypeKeys.RECEIVE_NEW_EXERCISES,
      },
    )).toEqual(newExercises);
  });
});
