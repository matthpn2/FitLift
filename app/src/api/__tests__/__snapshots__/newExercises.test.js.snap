// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async action creator tests should fetch current exercises 1`] = `
Array [
  Object {
    "type": "REQUEST_NEW_EXERCISES",
    "user": "SAMPLE_USER",
  },
  Object {
    "data": Object {
      "SAMPLE_USER": Object {
        "-L5GDNmo8uZxet9S7mE2": Object {
          "reps": 12,
          "timeStamp": 1518407604,
          "type": "Bicep Curls",
        },
        "-L5GDeJnjbOj1SkQRJK8": Object {
          "reps": 5,
          "timeStamp": 1518407604,
          "type": "Shoulder Press",
        },
        "-L5GDeeBKCPlTnm-yCiX": Object {
          "reps": 6,
          "timeStamp": 1518407604,
          "type": "Lateral Raises",
        },
      },
    },
    "type": "RECEIVE_NEW_EXERCISES",
  },
]
`;
